src/GenerateTemplateFiles.ts

```diff
@@ -291,6 +291,19 @@ export default class GenerateTemplateFiles {
   ): Promise<string[]> {
     const outputtedFilesAndFolders: string[] = [];
 
+    // Create a function to apply the transformations in one go
+    const regexEscape = (text: string) => text.replace(/([^a-zA-Z0-9_])/g, '\\$1');
+    const replacerLookup: Record<string, string> = {};
+    const replacerRegexBase = replacers
+      .map((replacer: IReplacer) => {
+        replacerLookup[replacer.slot] = replacer.slotValue;
+        return regexEscape(replacer.slot);
+      })
+      .join('|');
+    const replacerRegex = new RegExp(`^${replacerRegexBase}$`, 'g');
+    const replacer = (text: string) => text.replace(replacerRegex, (slot) => replacerLookup[slot]);
```

Comment:
Are you trying to remove `replace-string` as a dependency and write you own code? I don't understand the benefit to adding this code over the three lines that is already there:

```ts
replacers.forEach((replacer: IReplacer) => {
  output = replaceString(output, replacer.slot, replacer.slotValue);
});
```